### PR TITLE
[minor] Add MAS 8.6 to 8.7 upgrade roles

### PR DESCRIPTION
**Pull request by andrercm**
This PR addresses the upgrade for MAS 8.6 to 8.7 and its entire stack of dependencies (cert-manager, sbo, cpd, etc)

### CODE REVIEW

**Comment by andrercm** [comment]
@whitfiea glad you're back... i'm returning from vacation today but before i left, i had asked some help for code review this PR.
It's a very big effort, in summary this PR addresses the upgrade for MAS 8.6 to 8.7 and its entire stack of dependencies (cert-manager, sbo, cpd, etc)
This automates the steps described in this box notes: https://ibm.ent.box.com/notes/891391428610?s=rjyyxq0nenrb4rzbxui5fd8lryjo9ufe

**Comment by whitfiea** [comment]
@andrercm looks like a merge issue? The "Build Ansible collection" action is failing with:

```
Run yamllint --format standard -c $GITHUB_WORKSPACE/yamllint.yaml $GITHUB_WORKSPACE/ibm/mas_devops
/home/runner/work/ansible-devops/ansible-devops/ibm/mas_devops/roles/cp4d_db2wh/defaults/main.yml
  88:1      error    duplication of key "cpd_registry" in mapping  (key-duplicates)
  89:1      error    duplication of key "cpd_registry_user" in mapping  (key-duplicates)
  90:1      error    duplication of key "cpd_entitlement_key" in mapping  (key-duplicates)

/home/runner/work/ansible-devops/ansible-devops/ibm/mas_devops/roles/suite_dns/defaults/main.yaml
  74:1      error    duplication of key "cis_crn" in mapping  (key-duplicates)
  76:1      error    duplication of key "cis_subdomain" in mapping  (key-duplicates)
  81:1      error    duplication of key "cis_apikey" in mapping  (key-duplicates)
  84:1      error    duplication of key "cis_email" in mapping  (key-duplicates)
  86:1      error    duplication of key "cis_skip_dns_entries" in mapping  (key-duplicates)
  88:1      error    duplication of key "cis_skip_cluster_issuer" in mapping  (key-duplicates)
  90:1      error    duplication of key "update_dns" in mapping  (key-duplicates)
  92:1      error    duplication of key "custom_ocp_ingress" in mapping  (key-duplicates)
  100:1     error    duplication of key "custom_cluster_issuer" in mapping  (key-duplicates)
  104:1     error    duplication of key "custom_domain" in mapping  (key-duplicates)
```

File: ibm/mas_devops/roles/cert_manager_upgrade/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+suite_upgrade
```
Comment (whitfiea) [inline ibm/mas_devops/roles/cert_manager_upgrade/README.md:1]: looks like the wrong readme and should be for cert-manager_upgrade and not suite_upgrade

File: ibm/mas_devops/roles/cert_manager_upgrade_check/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+suite_upgrade
```
Comment (whitfiea) [inline ibm/mas_devops/roles/cert_manager_upgrade_check/README.md:1]: wrong readme

File: ibm/mas_devops/roles/cp4d_db2wh/defaults/main.yml [hunk 3]
```diff
@@ -21,15 +21,15 @@ db2wh_num_pods: "{{ lookup('env', 'DB2WH_NUM_PODS') | default(1, true) }}" # by
 
 # Configure meta storage for db2 (CP4D v3.5 and v4.0)
 db2wh_meta_storage_class: "{{ lookup('env', 'DB2WH_META_STORAGE_CLASS') }}"
-db2wh_meta_storage_size_gb: "{{ lookup('env', 'DB2WH_META_STORAGE_SIZE_GB') | default('20Gi', true) }}"
+db2wh_meta_storage_size_gb: "{{ lookup('env', 'DB2WH_META_STORAGE_SIZE_GB') | default('100Gi', true) }}"
```
Comment (whitfiea) [inline ibm/mas_devops/roles/cp4d_db2wh/defaults/main.yml:24]: What are the reasons for increasing the storage for these 3?

File: ibm/mas_devops/roles/cp4d_upgrade/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+---
+
+mas_instance_id: "{{ lookup('env', 'MAS_INSTANCE_ID') }}"
+mas_namespace: "mas-{{ mas_instance_id }}-core"
+mas_upgrade_supported_channels: ['maint86xdev', '8.6.x']
```
Comment (whitfiea) [inline ibm/mas_devops/roles/cp4d_upgrade/defaults/main.yml:5]: remove dev channel?

File: ibm/mas_devops/roles/ocp_upgrade/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+---
+# 1. (optionally) Upgrade OCP to v4.8
+# -----------------------------------------------------------------------------
+# MAS 8.7 is fully supported on both OCP 4.6, 4.8 (we can't support 4.7 because
+# CP4D does not), however at some point before you eventually upgrade to MAS
+# 8.8 you will need to upgrade OCP to v4.8
+
+# We can only really automate the upgrade in ROKS, in other cluster types
```
Comment (whitfiea) [inline ibm/mas_devops/roles/ocp_upgrade/tasks/main.yml:8]: Can we add a condition to the below to only upgrade if the cluster type is roks?

File: ibm/mas_devops/roles/suite_upgrade/defaults/main.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
+---
+
+mas_catalog_source: "{{ lookup('env', 'MAS_CATALOG_SOURCE') | default('ibm-operator-catalog', true) }}"
+mas_instance_id: "{{ lookup('env', 'MAS_INSTANCE_ID') }}"
+mas_namespace: "mas-{{ mas_instance_id }}-core"
+
+mas_upgrade_supported_channels: ['maint86xdev', '8.6.x']
```
Comment (whitfiea) [inline ibm/mas_devops/roles/suite_upgrade/defaults/main.yml:7]: remove dev channel?

File: ibm/mas_devops/roles/suite_upgrade_check/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+# 3. Check the existing installation
+# -----------------------------------------------------------------------------
+- name: "Check existing MAS installation"
+  include_tasks: tasks/check_mas_install.yml
+
+# # 4. Check installed applications are compatible with MAS 8.7
+# # -----------------------------------------------------------------------------
+# - name: "Check installed MAS application compatability with MAS 8.7"
+#   include_tasks: tasks/check_mas_apps.yml
```
Comment (whitfiea) [inline ibm/mas_devops/roles/suite_upgrade_check/tasks/main.yml:28]: We don't want to call this?

**Review by whitfiea — CHANGES_REQUESTED** [review]
Made some comments on the code. Generally the readmes need updating as these are all the same. Plus these new roles need to be added to https://github.com/ibm-mas/ansible-devops/blob/master/build/bin/copy-role-docs.sh and new playbooks https://github.com/ibm-mas/ansible-devops/tree/master/docs/playbooks, and both to the index https://github.com/ibm-mas/ansible-devops/blob/master/mkdocs.yml#L10

File: ibm/mas_devops/devplaybook.yml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
```
[the remark this answers is not in the document]
Comment (andrercm) [reply ibm/mas_devops/devplaybook.yml:1]: No, will remove it

File: ibm/mas_devops/playbooks/lite-iot-roks.yml [hunk 9]
```diff
@@ -25,18 +25,18 @@
 # We hope to find out how to avoid doing this entirely in the new year, as it
 # shouldn't be a requirement to restart worker nodes and modify the default image
 # pull secret to use any containerized software.
-- name: Update default cluster pull secret and reboot all worker nodes
-  import_playbook: cp4d/hack-worker-nodes.yml
-  vars:
-    cluster_type: roks
+# - name: Update default cluster pull secret and reboot all worker nodes
```
[the remark this answers is not in the document]
Comment (andrercm) [reply ibm/mas_devops/playbooks/lite-iot-roks.yml:28]: As far as i know we still need it

File: ibm/mas_devops/playbooks/lite-iot-roks.yml [hunk 10]
```diff
@@ -116,9 +116,45 @@
 - name: Install & configure IoT
   import_playbook: mas/install-app.yml
   vars:
+    mas_app_catalog_source: ibm-mas-iot-operators
```
[the remark this answers is not in the document]
Comment (andrercm) [reply ibm/mas_devops/playbooks/lite-iot-roks.yml:119]: yeah, removing all that now

File: ibm/mas_devops/roles/cp4d_db2wh/defaults/main.yml (hunk 3 above)
Comment (andrercm) [reply ibm/mas_devops/roles/cp4d_db2wh/defaults/main.yml:24]: By previous experience, using only 20Gi for db2wh system storage is not enough to install IoT + Manage in the same db... pretty quickly we start getting disk space issue on db2

File: ibm/mas_devops/roles/cp4d_upgrade/defaults/main.yml (hunk 4 above)
Comment (andrercm) [reply ibm/mas_devops/roles/cp4d_upgrade/defaults/main.yml:5]: yeah, i think at this point we can remove it safely.

File: ibm/mas_devops/roles/ocp_upgrade/tasks/main.yml (hunk 5 above)
Comment (andrercm) [reply ibm/mas_devops/roles/ocp_upgrade/tasks/main.yml:8]: We have an assertion for it here:
https://github.com/ibm-mas/ansible-devops/blob/upgrade87/ibm/mas_devops/roles/ocp_upgrade/tasks/ocp_upgrade.yml#L11

File: ibm/mas_devops/roles/suite_upgrade/defaults/main.yml (hunk 6 above)
Comment (andrercm) [reply ibm/mas_devops/roles/suite_upgrade/defaults/main.yml:7]: yeah, i think at this point we can remove it safely.

**Comment by andrercm** [comment]
@whitfiea I believe I have resolved all remaining requests for review, including docs. 
Please have another look and let me know if you think it is ready now to be merged.

**Review by whitfiea — APPROVED** [review]
Thanks for making the changes

**Review by durera — APPROVED** [review]

**Pull request merged**